src/plugin/v2/grpc_plugin.ts

```diff
@@ -0,0 +1,207 @@
+import * as grpc from 'grpc';
+import * as protoLoader from '@grpc/proto-loader';
+import * as promClient from 'prom-client';
+import { GrpcPlugin } from '../../node-plugin';
+import { Logger } from '../../logger';
+import { PluginConfig } from '../../config';
+import { createBrowser, Browser } from '../../browser';
+import { RenderOptions, HTTPHeaders } from '../../browser/browser';
+import {
+  RenderRequest,
+  RenderResponse,
+  CheckHealthRequest,
+  CheckHealthResponse,
+  CollectMetricsRequest,
+  CollectMetricsResponse,
+  HealthStatus,
+} from './types';
+
+const rendererV2PackageDef = protoLoader.loadSync(__dirname + '/../../../proto/rendererv2.proto', {
+  keepCase: true,
+  longs: String,
+  enums: String,
+  defaults: true,
+  oneofs: true,
+});
+
+const pluginV2PackageDef = protoLoader.loadSync(__dirname + '/../../../proto/pluginv2.proto', {
+  keepCase: true,
+  longs: String,
+  enums: String,
+  defaults: true,
+  oneofs: true,
+});
+
+const rendererV2ProtoDescriptor = grpc.loadPackageDefinition(rendererV2PackageDef);
+const pluginV2ProtoDescriptor = grpc.loadPackageDefinition(pluginV2PackageDef);
+
+export class RenderGRPCPluginV2 implements GrpcPlugin {
+  constructor(private config: PluginConfig, private log: Logger) {
+    populateConfigFromEnv(this.config);
+  }
+
+  async grpcServer(server: grpc.Server) {
+    const metrics = setupMetrics();
+    const browser = createBrowser(this.config.rendering, this.log);
+    const pluginService = new PluginGRPCServer(browser, this.log, metrics);
+
+    const rendererServiceDef = rendererV2ProtoDescriptor['pluginextensionv2']['Renderer']['service'];
+    server.addService(rendererServiceDef, pluginService);
+
+    const pluginServiceDef = pluginV2ProtoDescriptor['pluginv2']['Diagnostics']['service'];
+    server.addService(pluginServiceDef, pluginService);
+
+    metrics.up.set(1);
+
+    let browserVersion = 'unknown';
+
+    try {
+      browserVersion = await browser.getBrowserVersion();
+      metrics.browserInfo.labels(browserVersion).set(1);
+      this.log.debug('Using browser version', 'browserVersion', browserVersion);
+    } catch {
+      this.log.error('Failed to get browser version');
+      metrics.browserInfo.labels('unknown').set(0);
+    }
```

Comment:
As little as possible should be in a try block:
```suggestion
    let browserVersion = 'unknown';
    let labelValue = 1;

    try {
      browserVersion = await browser.getBrowserVersion();
    } catch (err) {
      this.log.error('Failed to get browser version', 'err', err);
      labelValue = 0;
    }
    metrics.browserInfo.labels(browserVersion).set(labelValue);
    if (browserVersion !== 'unknown') {
          this.log.debug('Using browser version', 'browserVersion', browserVersion);
    }
```